### PR TITLE
Add a review and confirmation page for substitution

### PR DESCRIPTION
**Pull request by eileen-nava**
Resolves [Jira issue 1082](https://vajira.max.gov/browse/CASEFLOW-1082)

### Description

This PR adds a review and confirm page where the user can confirm that they have selected a substitute appellant.  The user will be able to see the appellant’s information and information the tasks identified for creation.   This PR is dependent on PR #16192 

Testing Plan
- Navigate to a dispatched appeal where the appeal’s veteran has a date of death
- Create a substitute appeal and ensure that the new appellant’s information (name, relationship, and substitution date) display correctly
- Also, ensure that all automated tests pass

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] the user is able to see the following on the confirmation page: 
- Appellant Name 
- Relationship to Veteran
- Granted Substitution Date 
- Tasks identified for creation (if applicable)
- [ ] if an evidence submission task was selected, the window end date should be calculated with the following formula
- Original Evidence Submission Window End Date (actual) - Appellant Date of Death = number of days remaining + Granted Substitution Date = substitution evidence submission end date

### Testing Plan
- Navigate to a dispatched appeal where the appeal’s veteran has a date of death
- Create a substitute appeal and ensure that the new appellant’s information (name, relationship, and substitution date) display correctly
- Also, ensure that all automated tests pass

### CODE REVIEW

File: client/app/queue/substituteAppellant/tasks/utils.js [hunk 1]
```diff
@@ -1,6 +1,13 @@
...
+
 // Generic function to determine if a task (`current`) is a descendent of another task (`target`)
 // allItems is object keyed to a specified id
-export const isDescendant = (allItems = {}, target, current, { id = 'taskId' } = {}) => {
+export const isDescendant = (
+  allItems = {},
+  target,
+  current,
+  { id = 'taskId' } = {}
+) => {
```
Comment (n1zyy) [inline client/app/queue/substituteAppellant/tasks/utils.js:10]: chef's kiss

**Review by n1zyy — APPROVED** [review]
LGTM! :shipit:

File: client/app/queue/substituteAppellant/review/SubstituteAppellantReview.jsx [hunk 2]
```diff
@@ -8,15 +8,102 @@ import {
...
-export const SubstituteAppellantReview = ({ onBack, onCancel, onSubmit }) => {
   return (
     <>
       <AppSegment filledBackground>
         <section className={pageHeader}>
           <h1>{SUBSTITUTE_APPELLANT_REVIEW_TITLE}</h1>
           <div>{SUBSTITUTE_APPELLANT_REVIEW_SUBHEAD}</div>
         </section>
+        <section className={styles.tableSection}>
+          <h2>About the appellant</h2>
```
[the remark this answers is not in the document]
Comment (jcq) [reply client/app/queue/substituteAppellant/review/SubstituteAppellantReview.jsx:59]: Hah — don't get me started re usage (or lack thereof) of title case. 😅 

I think it's title case on the card header within Case Details, but designs had this lowercase, along with the other section headers. I prefer the former, but ah well.

File: client/app/queue/substituteAppellant/review/SubstituteAppellantReview.jsx [hunk 3]
```diff
@@ -8,15 +8,102 @@ import {
...
+  evidenceSubmissionEndDate,
+  relationship,
+  onBack,
+  onCancel,
+  onSubmit,
+}) => {
+  const substitutionDate = format(
+    parseISO(existingValues.substitutionDate),
+    'MM/dd/yyyy'
+  );
```
[the remark this answers is not in the document]
Comment (jcq) [reply client/app/queue/substituteAppellant/review/SubstituteAppellantReview.jsx:49]: Lol. I didn't even know that `DateString` existed. 😆 

I don't like its dependence upon `moment`, but that's sort of a Caseflow-wide issue. I think it's fine either way, so I wouldn't sweat it at all.

**Pull request merged**